src/playlist/playlistheader.cpp

```diff
@@ -71,6 +79,9 @@ void PlaylistHeader::contextMenuEvent(QContextMenuEvent* e) {
     QString title(
         model()->headerData(menu_section_, Qt::Horizontal).toString());
     hide_action_->setText(tr("&Hide %1").arg(title));
+    
+    // show rating_lock action only for ratings section
+    rating_lock_->setVisible(title == "Rating")
```

Comment:
I think `menu_section_` corresponds to `Playlist::Column` so you should use that instead of the string title, e.g.
```c++
rating_lock_->setVisible(menu_section_ == Playlist::Column_Rating)
```
but make sure it actually works.